### PR TITLE
DEVEXP-95: View recommendations and validation errors for certain sink config properties

### PR DESCRIPTION
**Pull request by rahulvudutala**
For a property with recommenders and validators, when the ConfigDef.NO_DEFAULT_VALUE is used as the default value, the property is becoming a required property in the ConfigDef. So I used an empty string as the default value.

### CODE REVIEW

File: src/main/java/com/marklogic/kafka/connect/MarkLogicConfig.java
Comment (BillFarber) [inline src/main/java/com/marklogic/kafka/connect/MarkLogicConfig.java:35]: For free-form String fields, I think we can just use the Validators that are included in the Kafka library.  Please take a look at the Validator for DSL_PLAN in MarkLogicSourceConfig.

File: src/main/java/com/marklogic/kafka/connect/MarkLogicConfig.java
Comment (BillFarber) [inline src/main/java/com/marklogic/kafka/connect/MarkLogicConfig.java:113]: Using the Validators that are included in the Kafka library will remove the necessity for this section of code.

File: src/main/java/com/marklogic/kafka/connect/MarkLogicConfig.java
Comment (BillFarber) [inline src/main/java/com/marklogic/kafka/connect/MarkLogicConfig.java:42]: Check out the Kafka built-in Validator, ConfigDef.Range. Look in MarkLogicSourceConfig for an example.

File: src/test/java/com/marklogic/kafka/connect/BuildDatabaseClientConfigTest.java
Comment (rjrudin) [inline src/test/java/com/marklogic/kafka/connect/BuildDatabaseClientConfigTest.java:164]: This looks like it should be broken up into several smaller tests. The test originally had a scope of verifying that invalid authentication was handled. The new test is much more difficult to understand since there are so many lines of code in it. Think it'd be much better and easier to understand if there's a test per config option - e.g. testInvalidDocumentFormat.

File: src/test/java/com/marklogic/kafka/connect/sink/HandleWriteFailureTest.java
Comment (rjrudin) [inline src/test/java/com/marklogic/kafka/connect/sink/HandleWriteFailureTest.java:27]: Is this implying that "json" is no longer valid? That seems like it would be a breaking change. Lowercase versions of these values should still be accepted.

**Review by rjrudin — CHANGES_REQUESTED** [review]

File: src/main/java/com/marklogic/kafka/connect/MarkLogicConfig.java
Comment (rjrudin) [inline src/main/java/com/marklogic/kafka/connect/MarkLogicConfig.java:37]: This should be in the sink config class, along with the other sink-specific validators.

**Review by rjrudin — CHANGES_REQUESTED** [review]
Looks good, just a little moving around needed

**Review by BillFarber — APPROVED** [review]
LGTM, but let's wait for an LGTM from @rjrudin before merging.

**Review by rjrudin — APPROVED** [review]
I was waiting on a LGTM from Peebs.

**Pull request merged**
